lib/state/revision/reducer.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	REVISION_IS_VIEWING,
+	REVISION_SELECT,
+} from '../action-types';
+
+export const isViewingRevisions = ( state = false, { type, isViewing } ) =>
+	REVISION_IS_VIEWING === type ? isViewing : state;
+
+export const selectedRevision = ( state = null, { type, revision } ) =>
+	REVISION_SELECT === type ? revision : state;
+
+export default combineReducers( {
+	isViewingRevisions,
```

Comment:
actually. isn't this kind-a related to the `isVisible` collection we have used for the other dialogs and modals?

it seems to me like an independently viewable dialog that could potentially have different visual designs while still keeping the prime notion of `visible` or `hidden`.

on the contrary, it does _not_ make sense to have it `visible` if the note editor is hidden, which would mean we have encoded an impossible state into our app.

maybe we could still go back to using one value to represent both things:

**open the revision slider**
 - issue `OPEN_REVISION_SLIDER`
 - something intercepts that action, loads in the revisions, then finds the newest one
 - that something issues a new `SELECT_REVISION` action to select the newest revision
 - reducer sets non-`null` value for `selectedRevision` indicating that slider should display

**cancel the revision selection**
 - issue `CLOSE_REVISION_SLIDER`
 - reducer sets `null` value for `selectedRevision` indicating that slider should hide

**select a revision from the slider**
 - issue `SELECT_REVISION`
 - reducer sets non-`null` value for `selectedRevision`, slider stays open

**close slider, selecting revision**
 - issue `SELECT_REVISION`
 - reducer sets non-`null` value for `selectedRevision`, slider stays open
 - issue `CLOSE_REVISION_SLIDER`
 - reducer sets `null` value for `selectedRevision` indicating that slider should hide